examples/performance/include/traccc/efficiency/seeding_performance_writer.hpp

```diff
@@ -70,6 +61,16 @@ class seeding_performance_writer {
         }
     }
 
+    void init() {
+        auto path = m_cfg.file_path;
+
+        m_output_file = TFile::Open(path.c_str(), m_cfg.file_mode.c_str());
```

Comment:
I only now realised that the memory management of this file is far from great. :frowning:

Please declare the file variable like:

```c++
std::unique_ptr<TFile> m_output_file;
```

That will make sure that we wouldn't leak memory. Which we currently do. (Calling `m_output_file->Close()` is not the same as deleting the object from memory.)

It doesn't necessarily have to happen in this PR, but it might as well.

Reply:
Yeah thanks for the comments. I used the `unique_ptr` for `TFile` now